varianteval/core/io/vcf.py

```diff
@@ -53,9 +61,136 @@ def vcf2callset(vcf_fname, filter_funcs=None):
+    :param str: the ALT field of a BND record of a VCF file.
+    """
+    q: int = string.find(":")
+    p: int = string.find("[")
+    if p == -1: p = string.find("]")
+    return string[p+1:q]
+
+
+def get_breakend_chromosome_position(string: str) -> int:
+    """
+    :param str: the ALT field of a BND record of a VCF file.
+    """
+    q: int = string.find(":")
+    p: int = string.rfind("[")
+    if p < 0: p = string.rfind("]")
+    return int(string[q+1:p]) - 1  # VCF positions are one-based
+
+
+def get_breakend_chromosome_side(string: str, position_id: bool) -> int:
+    """
+    :param string: the ALT field of a BND record of a VCF file;
+    :param position_id: TRUE=first position in the VCF record (encoded by the 
+    CHROM and POS fields); FALSE=second position in the VCF record (encoded by 
+    the ALT field).
+    :return: value defined in ``constants.BREAKPOINT_SIDE``.
+    """
+    if position_id:
+        if string[0] == "[" or string[0] == "]":
+            return BREAKPOINT_SIDE.LEFT
+        else:
+            return BREAKPOINT_SIDE.RIGHT
+    else
+        if string.find("[") >= 0:
+            return BREAKPOINT_SIDE.LEFT
+        else:
+            return BREAKPOINT_SIDE.RIGHT
+
+
+def get_confidence_interval(record: VariantRecord, is_first_pos: bool, sigma_multiple: int) -> tuple[int, int]:
+    """
+    Returns the range of uncertainty around a position ``X`` (implicit from the 
+    context).
+    
+    :param is_first_pos: TRUE=first position in a VCF record; FALSE=last 
+    position;
+    :param sigma_multiple: if the record does not contain coordinates but just 
+    the STD, the procedure assumes an interval that extends this multiple of the
+    STD on both sides (2 or 3 captures most of a Gaussian);
+    :return: First element: quantity to be added to ``X`` to get the first 
+    position of the uncertainty interval (typically negative or zero). Second 
+    element: quantity to be added to ``X`` to get the last position of the 
+    uncertainty interval (typically positive or zero).
+    """
+    quantum: int
+    value: float	
+
+    if is_first_pos:
+        if VCF_CIPOS_STR in record.info.keys():
+            return tuple(map(int, record.info.get(VCF_CIPOS_STR).split(VCF_CI_SEPARATOR)))
+            # The sign is already ok
+        for string in VCF_STD_START:
+            if string in record.info.keys():
+                value = float(record.info.get(string))
+                quantum = int(value*SIGMA_MULTIPLE)
+                return (-quantum, quantum)
+    else:
+        if VCF_CIEND_STR in record.info.keys():
+            return tuple(map(int, record.info.get(VCF_CIEND_STR).split(VCF_CI_SEPARATOR)))
+            # The sign is already ok
+        if VCF_CILEN_STR in record.info.keys():
+            return tuple(map(int, record.info.get(VCF_CILEN_STR).split(VCF_CI_SEPARATOR)))
+            # The sign is already ok
+        for string in VCF_STD_END:
+            if string in record.info.keys():
```

Comment:
nit: this block is a copy of the block above; the logic could be factored out to condition on is_first_pos for the constant to inspect (e.g. fields = VCF_STD_START if is_first_pos else VCF_STD_END)